### PR TITLE
[user model] changes and fixes made during testing (round 1)

### PR DESCRIPTION
**Pull request by nan-li**
# Description
## One Line Summary
An unfortunately long and mixed hodge podge of changes during testing.

### Motivation
Testing

# Testing
## Unit testing
None

## Manual testing
Lots, on physical iPhone 13

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [ ] I have filled out all **REQUIRED** sections above
   - [ ] PR does one thing
   - [ ] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [ ] I have included test coverage for these changes, or explained why they are not needed
   - [ ] All automated tests pass, or I explained why that is not possible
   - [ ] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [ ] Code is as readable as possible.
   - [ ] I have reviewed this PR myself, ensuring it meets each checklist item

- - -
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1192)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift [hunk 1]
```diff
@@ -713,28 +835,42 @@ class OSRequestTransferSubscription: OneSignalRequest, OSUserRequest {
...
         var subscriptionParams = subscriptionObject
         subscriptionParams.removeValue(forKey: "address")
-        subscriptionParams["token"] = subscriptionObject["address"]
-
+        subscriptionParams.removeValue(forKey: "notificationTypes")
+        subscriptionParams["token"] = subscriptionModel.address ?? ""
+        subscriptionParams["notification_types"] = subscriptionModel.notificationTypes
+        subscriptionParams["enabled"] = subscriptionModel.enabled
+        // TODO: The above is not quite right. If we hydrate, we will over-write any pending updates
+//        subscriptionParams["token"] = subscriptionObject["address"]
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift:871]: Remove commented out code if not needed.

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift (hunk 1 above)
Comment (Dodalun) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift:871]: > Remove commented out code if not needed.

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift [hunk 2]
```diff
@@ -60,29 +61,63 @@ class OSUserExecutor {
...
+            if let response = response {
+                parseFetchUserResponse(response: response, identityModel: request.identityModel)
             }
-
-            // hydrate identity, properties, subscriptions
-
             executePendingRequests()
-        } onFailure: { _ in
+        } onFailure: { error in
+            print("ECM test + \(error.debugDescription)")
```
[the remark this answers is not in the document]
Comment (nan-li) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift:120]: Done. There will be a followup PR where all the misc `print()`s will be moved to `onesignalLog` with better message.

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift [hunk 3]
```diff
@@ -188,36 +225,50 @@ protocol OSUserRequest: OneSignalRequest, NSCoding {
...
+        self.stringDescription = "OSRequestCreateUser"
 
         super.init()
 
         var pushSubscriptionObject: [String: Any] = [:]
         pushSubscriptionObject["id"] = pushSubscriptionModel.subscriptionId
         pushSubscriptionObject["type"] = pushSubscriptionModel.type.rawValue
-        pushSubscriptionObject["token"] = pushSubscriptionModel.address
-        // ... and more ? ...
+        pushSubscriptionObject["token"] = pushSubscriptionModel.address ?? "test" + String(arc4random())
```
[the remark this answers is not in the document]
Comment (nan-li) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift:257]: Done

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift (hunk 1 above)
Comment (nan-li) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift]: Done, replaced with better comment to address the TODO mentioned.

Comment (Dodalun) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift]: image

**Review by Dodalun — APPROVED** [review]
👌😁

**Review by emawby — APPROVED** [review]

**Review by fhboswell — APPROVED** [review]

**Pull request merged**